dashboard/api_edx_cache_test.py

```diff
@@ -0,0 +1,306 @@
+        course_id = list(self.enrollment_ids)[0]
+        enrollment = self.enrollments.get_enrollment_for_course(course_id)
+        self.assert_cache_in_db()
+
+        # normal update that creates also the entry
+        CachedEdxUserData.update_cached_enrollment(self.user, enrollment, course_id, False)
+        self.assert_cache_in_db(enrollment_keys=[course_id])
+        cached_enr = CachedEnrollment.objects.get(user=self.user, course_run__edx_course_key=course_id)
+        assert cached_enr.data == enrollment.json
+        assert mocked_index.delay.called is False
+        # update of different data with indexing
+        enr_json = {
+            "course_details": {
+                "course_id": course_id,
+            },
+            "is_active": True,
+            "mode": "verified",
+            "user": self.user.username
+        }
+        enrollment_new = Enrollment(enr_json)
+        CachedEdxUserData.update_cached_enrollment(self.user, enrollment_new, course_id, True)
+        self.assert_cache_in_db(enrollment_keys=[course_id])
+        cached_enr.refresh_from_db()
+        assert cached_enr.data == enr_json
+        assert mocked_index.delay.called is True
+
+    @patch('search.tasks.index_users', autospec=True)
+    def test_update_cached_enrollments(self, mocked_index):
+        """Test for update_cached_enrollments."""
+        self.assert_cache_in_db()
+        assert UserCacheRefreshTime.objects.filter(user=self.user).exists() is False
+        CachedEdxUserData.update_cached_enrollments(self.user, self.edx_client)
+        self.assert_cache_in_db(enrollment_keys=self.enrollment_ids)
+        cache_time = UserCacheRefreshTime.objects.get(user=self.user)
+        now = datetime.now(tz=pytz.UTC)
+        assert cache_time.enrollment <= now
+        assert mocked_index.delay.called is True
+        mocked_index.reset_mock()
+
+        # add another cached element for another course that will be removed by the refresh
+        cached_enr = CachedEnrollmentFactory.create(user=self.user)
+        self.assert_cache_in_db(enrollment_keys=list(self.enrollment_ids) + [cached_enr.course_run.edx_course_key])
+        CachedEdxUserData.update_cached_enrollments(self.user, self.edx_client)
+        self.assert_cache_in_db(enrollment_keys=self.enrollment_ids)
+        cache_time.refresh_from_db()
+        assert cache_time.enrollment >= now
+        assert mocked_index.delay.called is True
+
+    @patch('search.tasks.index_users', autospec=True)
+    def test_update_cached_certificates(self, mocked_index):
+        """Test for update_cached_certificates."""
+        assert len(self.certificates_ids) > len(self.verified_certificates_ids)
+        self.assert_cache_in_db()
+        assert UserCacheRefreshTime.objects.filter(user=self.user).exists() is False
+        CachedEdxUserData.update_cached_certificates(self.user, self.edx_client)
+        self.assert_cache_in_db(certificate_keys=self.verified_certificates_ids)
```

Comment:
`self.assert_cache_in_db(certificate_keys=self.verified_certificates_ids)`
verifies that ONLY the passed ids are in the cache. This together with the first call (and the mocked responses) should answer your question.
If you want I can make an explicit query for this.